src/tests/nativeaot/SmokeTests/PInvoke/PInvokeNative.cpp

```diff
@@ -529,14 +540,18 @@ DLL_EXPORT void __stdcall StructTest_ByOut(NativeSequentialStruct *nss)
 
     int arrSize = 7;
     char *p;
-    p = (char *)MemAlloc(sizeof(char) * arrSize);
+    p = (char *)MemAlloc(sizeof(char) * arrSize + 1);
```

Comment:
This is unrelated. I almost copy pasted this so I looked at it. We were writing the null terminator past the end of the buffer...